spec/helper.js

```diff
@@ -182,11 +191,17 @@ beforeEach(async () => {
 });
 
 afterEach(function (done) {
-  const afterLogOut = () => {
+  const afterLogOut = async () => {
     if (Object.keys(openConnections).length > 0) {
       fail('There were open connections to the server left after the test finished');
     }
-    TestUtils.destroyAllDataPermanently(true).then(done, done);
+    await TestUtils.destroyAllDataPermanently(true);
+    if (didChangeConfiguration) {
```

Comment:
That's a clever idea, but doesn't that cause an unnecessary reconfiguration in many cases? 

I'm thinking:
- Test A does reconfiguration with custom config
- afterEach does reconfiguration with default config
- Test B does reconfiguration with custom config

Reply:
Test A does reconfiguration with custom config
afterEach does reconfiguration with default config
Test B doesn't do reconfiguration
afterEach doesn't do reconfiguration
Test C doesn't do reconfiguration
afterEach doesn't do reconfiguration
Test D does reconfiguration with custom config

The net reconfiguration is the same. If you are interested I left a counter in the test suite

https://github.com/parse-community/parse-server/runs/2099411665

As you can see it reconfigures 461 times. Which is much better than the 2825 it was before.

Reply:
I think I'm getting it. You are assuming that the tests B and C do not need a reconfig, therefore none should be done. And removing these unnecessary reconfigs, was something this PR also does (among other things)?

Reply:
Yes

Reply:
I figured out how random testing works. 

1) Randomize the test suites and run them.
2) Before a test suite runs randomize the test within them.

Basically randomizing a single test suite can isolate the problem instead of running the entire suite.

I'll try it in a separate PR. Only 40 failing tests to go.

Reply:
Great discovery! And just to be clear, no tests were removed or disabled in this PR, correct? I noticed re-ordering and re-grouping, but not sure whether I've missed something else.

Reply:
That is correct